naming/src/main/java/org/jboss/as/naming/subsystem/NamingBindingResourceDefinition.java

```diff
@@ -56,6 +57,8 @@
  */
 public class NamingBindingResourceDefinition extends SimpleResourceDefinition {
 
+    private static final String[] ALLOWED_TYPES = {"char", "java.lang.Character", "byte", "java.lang.Byte", "short", "java.lang.Short", "int", "java.lang.Integer", "long", "java.lang.Long", "float", "java.lang.Float", "double", "java.lang.Double", "boolean", "java.lang.Boolean", "java.lang.String", "java.net.URL"};
```

Comment:
Any reason why not to use  `java.lang.Class#getName` e.g. `Character.class.getName()`, etc?

Reply:
This list is copied from `NamingBindingAdd.coerceToType` method, but yes, we can use that approach as well.